core/arch/arm/kernel/generic_boot.c

```diff
@@ -551,6 +554,44 @@ static int add_optee_res_mem_dt_node(void *fdt)
 	return 0;
 }
 
+static void read_optee_secure_device_node(void *fdt)
+{
+	const uint32_t *p;
+	int offs;
+	int len;
+#if (TRACE_LEVEL >= TRACE_INFO)
+	char s[33];
+	int n = 0;
+	int m;
+#endif
+
+	offs = fdt_path_offset(fdt, "/firmware/optee");
+	if (offs < 0)
+		return;
+
+	p = fdt_getprop(fdt, offs, "secure-device-id", &len);
+	if (!p || !len)
+		return;
+
+	if (len > (int)sizeof(secure_device_id))
+		len = sizeof(secure_device_id);
+
+	memcpy(secure_device_id, p, len);
+	secure_device_id_len = len;
+
```

Comment:
Here I think you should scrub the `secure-devide-id` node in the DT. Because you probably don't want to leak this information to the NS world (I'm assuming the ID is read by ARM-TF from a protected area -- which is not the case if ID is the eMMC CID of course, but I'm thinking more generally).